src/helpers/internal_viewport.ts

```diff
@@ -143,53 +143,55 @@ export class InternalViewport {
+  adjustPositionX(targetCol: HeaderIndex) {
     const sheetId = this.sheetId;
-    const { start, end } = this.getters.getColDimensions(sheetId, col);
-    while (
-      end > this.offsetX + this.offsetCorrectionX + this.width &&
-      this.offsetX + this.offsetCorrectionX < start
-    ) {
-      this.offsetX = this.getters.getColDimensions(sheetId, this.left).end - this.offsetCorrectionX;
-      this.offsetScrollbarX = this.offsetX;
-      this.adjustViewportZoneX();
-    }
-    while (col < this.left) {
-      let leftCol: HeaderIndex;
-      for (leftCol = this.left; leftCol >= 0; leftCol--) {
-        if (!this.getters.isColHidden(sheetId, leftCol)) {
-          break;
-        }
+    const { end } = this.getters.getColDimensions(sheetId, targetCol);
+    const maxCol = this.getters.getNumberCols(sheetId);
+
+    if (this.offsetX + this.offsetCorrectionX + this.width < end) {
+      for (let col = this.left; this.offsetX + this.offsetCorrectionX + this.width < end; col++) {
+        if (col > maxCol) break;
+        if (this.getters.isColHidden(sheetId, col)) continue;
+
+        this.offsetX = this.getters.getColDimensions(sheetId, col).end - this.offsetCorrectionX;
+        this.offsetScrollbarX = this.offsetX;
+        this.adjustViewportZoneX();
+      }
+    } else if (this.left > targetCol) {
+      for (let col = this.left; col >= targetCol; col--) {
+        if (col < 0) break;
+        if (this.getters.isColHidden(sheetId, col)) continue;
+
+        this.offsetX = this.getters.getColDimensions(sheetId, col).start - this.offsetCorrectionX;
+        this.offsetScrollbarX = this.offsetX;
+        this.adjustViewportZoneX();
       }
-      this.offsetX =
-        this.getters.getColDimensions(sheetId, leftCol - 1).start - this.offsetCorrectionX;
-      this.offsetScrollbarX = this.offsetX;
-      this.adjustViewportZoneX();
     }
   }
 
-  adjustPositionY(row: HeaderIndex) {
+  adjustPositionY(targetRow: HeaderIndex) {
     const sheetId = this.sheetId;
-    while (
-      this.getters.getRowDimensions(sheetId, row).end >
-        this.offsetY + this.height + this.offsetCorrectionY &&
-      this.offsetY + this.offsetCorrectionY < this.getters.getRowDimensions(sheetId, row).start
-    ) {
-      this.offsetY = this.getters.getRowDimensions(sheetId, this.top).end - this.offsetCorrectionY;
-      this.offsetScrollbarY = this.offsetY;
-      this.adjustViewportZoneY();
-    }
-    while (row < this.top) {
-      let topRow: HeaderIndex;
-      for (topRow = this.top; topRow >= 0; topRow--) {
-        if (!this.getters.isRowHidden(sheetId, topRow)) {
-          break;
-        }
+    const { end } = this.getters.getRowDimensions(sheetId, targetRow);
+    const maxRow = this.getters.getNumberRows(sheetId);
+
+    if (this.offsetY + this.height + this.offsetCorrectionY < end) {
+      for (let row = this.top; this.offsetY + this.height + this.offsetCorrectionY < end; row++) {
+        if (row > maxRow) break;
```

Comment:
maybe write this with some brackets, actually. I know it seems like wasted space/lines but I'd argue that the brackets are
- more readable in general
- less error prone (see https://www.imperialviolet.org/2014/02/22/applebug.html as a nice example)
- less likely to break upon future changes

I feel it's worth having a guideline over the whole code base for this

Reply:
Personnaly, I think that no bracket makes the code a bit easier to read, and not error prone _as long as you can write everything on a single line_. If you have a line break with prettier, it becomes error prone I agree. 

But we could use a guideline for the whole code base yeah. Idk how @LucasLefevre feels about it.

Reply:
I do like my one liners 😢 But if you both perfer brackets, I can take the habit of puttin them.